src/main/java/com/amazon/ion/impl/macro/MacroEvaluatorAsIonReader.kt

```diff
@@ -35,8 +35,11 @@ class MacroEvaluatorAsIonReader(
     private fun queueNext() {
         queuedValueExpression = null
         while (queuedValueExpression == null) {
-            val nextCandidate = evaluator.expandNext() ?: return
-            when (nextCandidate) {
+            when (val nextCandidate = evaluator.expandNext()) {
+                null -> {
+                    queuedFieldName = null
+                    return
+                }
```

Comment:
🗺️ The only functional difference here is that the queued field name is cleared so that we don't have any chance of the field name leaking between containers. `MacroEvaluator` only returns `null` if it's at the end of a container or the end of the top-level of the expansion, so the field name should always be cleared if it returns null.